cardano-db-sync/src/Cardano/DbSync/DbAction.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+
+module Cardano.DbSync.DbAction
+  ( DbAction (..)
+  , DbActionQueue (..)
+  , MkDbAction (..)
+  , blockingFlushDbActionQueue
+  , lengthDbActionQueue
+  , newDbActionQueue
+  , writeDbActionQueue
+  ) where
+
+import           Cardano.Prelude
+
+import           Cardano.DbSync.Types
+
+import qualified Control.Concurrent.STM as STM
+import           Control.Concurrent.STM.TBQueue (TBQueue)
+import qualified Control.Concurrent.STM.TBQueue as TBQ
+
+import           Ouroboros.Consensus.Byron.Ledger (ByronBlock (..))
+import           Ouroboros.Network.Block (Point (..), Tip)
+
+
+data DbAction
+  = DbApplyBlock !CardanoBlockTip
+  | DbRollBackToPoint !CardanoPoint
+  | DbFinish
```

Comment:
What is this for? "End of the world" scenario?

Reply:
It has been there for ages. It can probably be removed at some point, but I won't do it now because simple Shelley support and then full Shelly support is a priority,